drivers/display/mb_display.c

```diff
@@ -176,15 +176,15 @@ static inline void update_pins(struct mb_display *disp, u32_t val)
 	u32_t pin, prev = (disp->cur + 2) % 3;
 
 	/* Disable the previous row */
-	gpio_pin_write(disp->dev, ROW_PIN(prev), 0);
+	gpio_pin_set_raw(disp->dev, ROW_PIN(prev), 0);
 
 	/* Set the column pins to their correct values */
 	for (pin = LED_COL1_GPIO_PIN; pin <= LED_COL9_GPIO_PIN; pin++) {
-		gpio_pin_write(disp->dev, pin, !!(val & BIT(pin)));
+		gpio_pin_set_raw(disp->dev, pin, !!(val & BIT(pin)));
```

Comment:
I did not want to change the original `!!()`, leaving it to @jhedberg if he prefers to use a simple `&` instead.

Reply:
Aren't the correct values of the second parameter 0 and 1? That's what the !! is there for.

Reply:
The way the function is documented `x & mask` should work ("Writing any value other than 0 will set it to a high physical level.")  Either `!!(x & mask)` or `(x & mask) != 0` are good for clarity of intention.  Sometimes people are confused by `!!x` though I personally like it (except where MISRA essential type requires a "boolean" expression).